### PR TITLE
Analog Devices ADIS16470 driver for CUAV x7pro

### PR DESCRIPTION
**Pull request by dagar**
The driver is quite similar to the adis16477, but worth redoing for the newer structure.

### CODE REVIEW

**Comment by jinger26** [comment]
Test results on 3/12  by @CUAVcaijie :


**Test environment：（测试环境：）

Firmware version :（固件版本）
c356181f9066bc6361825b2ae17212aae01cd55f

Test hardware:（测试硬件）
CUAV X7Pro

Firmware modification ：（固件修改）
/Firmware/boards/cuav/x7pro/src/spi.cpp
DRV_DEVTYPE_UNUSED modified to DRV_IMU_DEVTYPE_ADIS16477 
image



**Test process：（测试流程）

Use QGC to send commands，Simultaneous use of logic analyzer to capture waveforms  : （使用QGC发送命令，同时使用逻辑分析仪捕获波形 ）
adis16477 -s start

image


The number of pulses of the clock is correct at 16, and it is tested to be 32 pulses （时钟的脉冲数在16才是正确的，经测试是32个脉冲）

image



**Result：（结果）
In the firmware version （c356181f9066bc6361825b2ae17212aae01cd55f） There is a problem with the SPI16 bit reading and writing

**Comment by dagar** [comment]
The ADIS16470 driver is now working on the CUAV X7 pro.

From the test rack.

Screen Shot 2021-03-16 at 4 01 39 PM

Screen Shot 2021-03-16 at 4 02 31 PM

Screen Shot 2021-03-16 at 4 02 49 PM

The driver's maximum output data rate (ODR) is 2000 Hz and is scheduled by the data ready pin.

Screen Shot 2021-03-16 at 4 03 29 PM

The minor difference is rate (2005 Hz vs 2000 Hz) is due to the separate clock in the ADIS16470. The difference is quite minor (~ 0.25%) and likely negligible, but could be further improved in the future by also connecting the SYNC pin.

**Review by LorenzMeier — DISMISSED** [review]

File: boards/cuav/x7pro/default.cmake [hunk 1]
```diff
@@ -66,8 +67,8 @@ px4_add_board(
 		flight_mode_manager
 		fw_att_control
 		fw_pos_control_l1
-		gyro_calibration
-		gyro_fft
+		#gyro_calibration
```
Comment (dagar) [inline boards/cuav/x7pro/default.cmake:70]:
```suggestion
		gyro_calibration
```

File: boards/cuav/x7pro/default.cmake [hunk 2]
```diff
@@ -66,8 +67,8 @@ px4_add_board(
...
+		#gyro_fft
```
Comment (dagar) [inline boards/cuav/x7pro/default.cmake:71]:
```suggestion
		gyro_fft
```

**Comment by CUAVcaijie** [comment]
> Test hardware:
CUAV X7Pro

> ADIS16470 test:
debug output:
image

> MAVLINK Inspector:
image
image
image

> mavlink shell:
image

> Result：
ADIS16470 can be detected normally, and the data is also normal

**Comment by davids5** [comment]
@CUAVcaijie - the proper fix for this is now pending in nuttx.  https://github.com/apache/incubator-nuttx/pull/3167

Please retest with test with https://github.com/PX4/PX4-Autopilot/pull/17213

**Pull request merged**